keyring/testing/backend.py

```diff
@@ -181,3 +181,20 @@ def test_new_with_properties(self):
         assert alt.foo == 'bar'
         with pytest.raises(AttributeError):
             self.keyring.foo  # noqa: B018
+
+    def test_wrong_username_returns_none(self):
+        keyring = self.keyring
+        service = 'test_wrong_username_returns_none'
+        cred = keyring.get_credential(service, None)
+        assert cred is None
+
+        password_1 = 'password1'
+        password_2 = 'password2'
+        self.set_password(service, 'user1', password_1)
+        self.set_password(service, 'user2', password_2)
+
+        assert keyring.get_credential(service, "user1").password == password_1
+        assert keyring.get_credential(service, "user2").password == password_2
+        
```

Comment:
Question/Note:
Considered adding this assertion as well (see [previous commit)](https://github.com/jaraco/keyring/pull/699/commits/27d0e0ea3ea75ad7400e9d1d42bcbbbe088aaa01) but it was failing intermittently on linux.

Sometimes `keyring.get_credential(service, None).password ` would equate to `password_2` and sometimes to `password_1`.

Sticking a `time.sleep(1) between the 2 `set_password` calls above helped it reliably pass so perhaps there's some async/timing thing happening with dbus (can possibly look into that deeper but a bit out of my depths w/ dbus)

```suggestion
        assert keyring.get_credential(service, "user2").password == password_2
         # Most recent entered
        assert keyring.get_credential(service, None).password == password_2
```

Reply:
Probably that additional assertion should be a separate test. Better to test one thing (the thing that matches the function's name). I'll use your comment to file a separate issue to track this concern.